### DISCUSSION TITLE
Kafka Subscription

### DISCUSSION THREAD

**Discussion by makp0** [Eventuous/eventuous#184] [Q&A] [answered]
KafkaBasicSubscription is not implemented atm.   
BasicProducerTests shows a consumption of messages produced by KafkaBasicProducer.

What are the challenges refraining KafkaBasicSubscription implementation?

**Comment by alexeyzimarev** [comment, marked as answer]
Basically, no issues as such, I just never got to it. https://github.com/Eventuous/eventuous/issues/91

**Reply by alexeyzimarev** [reply]
Also, the main limitation of the basic producer and, potentially, the subscription, is lack of schema registry support. The magic byte won't be there, so some consumers might just not be able to consume.

**Reply by makp0** [reply]
Thank you for a quick response!

So, atm if one wants to integrate eventuous with an on-premise message bus - should one use RabbitMQ?

**Reply by alexeyzimarev** [reply]
I guess it depends on what you want to use, or already use in-house. RMQ is indeed a popular option.
